sdk/cosmosdb/cosmos/src/client/Item/Items.ts

```diff
@@ -373,4 +383,68 @@ export class Items {
       ref
     );
   }
+
+  public async bulk(
+    operations: Operation[],
+    options?: RequestOptions
+  ): Promise<OperationResponse[]> {
+    const {
+      resources: partitionKeyRanges,
+    } = await this.container.readPartitionKeyRanges().fetchAll();
+    const { resource: definition } = await this.container.getPartitionKeyDefinition();
+    const batches: Batch[] = partitionKeyRanges.map((keyRange: PartitionKeyRange) => {
+      return {
+        min: keyRange.minInclusive,
+        max: keyRange.maxExclusive,
+        rangeId: keyRange.id,
+        indexes: [],
+        operations: [],
+      };
+    });
+    operations
+      .map((operation) => addPKToOperation(operation, definition))
+      .forEach((operation: Operation, index: number) => {
+        const partitionProp = definition.paths[0].replace("/", "");
+        const isV2 = definition.version && definition.version === 2;
+        const toHashKey = getPartitionKeyToHash(operation, partitionProp);
+        const hashed = isV2 ? hashV2PartitionKey(toHashKey) : hashV1PartitionKey(toHashKey);
+        const batchForKey = batches.find((batch: Batch) => {
+          return isKeyInRange(batch.min, batch.max, hashed);
+        });
+        batchForKey.operations.push(operation);
+        batchForKey.indexes.push(index);
+      });
+
+    const path = getPathFromLink(this.container.url, ResourceType.item);
+
+    const orderedResponses: OperationResponse[] = [];
+    await Promise.all(
+      batches
+        .filter((batch: Batch) => batch.operations.length)
+        .map(async (batch: Batch) => {
+          try {
+            const response = await this.clientContext.bulk({
+              body: batch.operations,
```

Comment:
Each service request can take 100 operations and be up to 2 MB (recommend ~200 KB unless a single doc is bigger which which case fit only one operation into the request).